### PR TITLE
LifeCycleAssessment_oM: MEP Scope Refactor

### LINKED ISSUE #1023: LifeCycleAssessment_oM: MEPScope object refactor
#### Description:

Includes four new scope objects within the MEP Scope: Mechanical, Electrical, Plumbing, and Fire Protection. 
Additional ObjectCategories and ObjectScopes have also been added to account for proposed object properties within each scope. The exact property names may change in upcoming workshops during this milestone, but this should provide a good foundation for further decisions within the MEP space.

### PR DESCRIPTION
**Pull request by michaelhoehn**
### Issues addressed by this PR

Closes #1023 

### Test files

[Test File](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/LifeCycleAssessment_Toolkit/LifeCycleAssessment_oM-%231023%20MEPScope%20refactor.gh?csf=1&web=1&e=T6Ln1K)

### CODE REVIEW

File: LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace BH.oM.LifeCycleAssessment
+{
+    [Description("Mechanical Scope")]
+    public class MechanicalScope : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        [Description("MEP Ductwork is a material (eg sheet metal) that helps to convey airflow from heating, ventilation or cooling systems")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs:36]: MEP Ductwork includes materials (eg sheet metal) that convey airflow from heating, ventilation or cooling systems

File: LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        public virtual List<Duct> Ducts { get; set; } = new List<Duct>();
+
+        [Description("MEP Equipment is a machine that processes mechanical, electrical or plumbing loads (eg Fan, Electrical Panel, Pump")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs:39]: Mechanical equipment includes machines and devices that provide heating, cooling, venting, pumping etc. (eg chillers, fans, AHUs, pumps")

File: LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        public virtual List<IBHoMObject> Equipment { get; set; } = new List<IBHoMObject>();
+        
+        [Description("MEP Piping is a material (eg copper) that helps to convey fluids (eg water, waste) within a building")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/MechanicalScope.cs:42]: Mechanical pipework includes distribution systems (eg copper) that convey fluids.

File: LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace BH.oM.LifeCycleAssessment
+{
+    [Description("Plumbing Scope.")]
+    public class PlumbingScope : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        [Description("MEP Equipment is a machine that processes mechanical, electrical or plumbing loads (eg Fan, Electrical Panel, Pump")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs:35]: Electrical equipment includes machines or devices that process electrical loads (eg electrical panels, transformers)

File: LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        public virtual List<IBHoMObject> Equipment { get; set; } = new List<IBHoMObject>();
+        
+        [Description("MEP Piping is a material (eg copper) that helps to convey fluids (eg water, waste) within a building")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs:38]: Plumbing pipework includes distribution systems (eg PVC) that convey fluids

File: LifeCycleAssessment_oM/ScopeObjects/FireProtectionScope.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    [Description("Fire Protection Scope.")]
+    public class FireProtectionScope : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        [Description("Fire protection equipment used to facilitate functionality of the overall system")]
+        public virtual List<IBHoMObject> Equipment { get; set; } = new List<IBHoMObject>();
+
+        [Description("BHoM Pipes used for fire protection")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/FireProtectionScope.cs:39]: Fireproofing pipework includes distribution systems that convey fluids in the case of fire

File: LifeCycleAssessment_oM/ScopeObjects/FireProtectionScope.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace BH.oM.LifeCycleAssessment
+{
+    [Description("Fire Protection Scope.")]
+    public class FireProtectionScope : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        [Description("Fire protection equipment used to facilitate functionality of the overall system")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/FireProtectionScope.cs:36]: Fire protection equipment includes machines or devices that comprise the system (eg smoke alarms)

Comment (kayleighhoude) [reply LifeCycleAssessment_oM/ScopeObjects/FireProtectionScope.cs:36]: @ajensen19 smoke alarms fall under a different discipline!***Fire protection equipment includes both fire and jockey pumps which convey water through the fire protection system

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace BH.oM.LifeCycleAssessment
+{
+    [Description("Electrical Scope.")]
+    public class ElectricalScope : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+        [Description("MEP Batties are energy storage devices (eg photovoltaic panels)")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:36]: Batteries include energy storage devices (eg lithium-ion batteries)

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> Batteries { get; set; } = new List<IBHoMObject>();
+
+        [Description("MEP Cable Trays used to route electrical wiring throughout a building")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:39]: Electrical cable trays are used to route electrical wiring throughout a building

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> CableTrays { get; set; } = new List<IBHoMObject>();
+
+        [Description("MEP Conduit is a tube used to route electrical wiring")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:42]: Electrical conduit includes tubing or channels which are used to route electrical wiring.

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> Conduit { get; set; } = new List<IBHoMObject>();
+
+        [Description("Electrical equipment used throughout the building")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:45]: Electrical equipment includes machines or devices that comprise the electrical infrastructure (meters, switchboards, transformers, solar panels)

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> Equipment { get; set; } = new List<IBHoMObject>();
+
+        [Description("MEP Generators are devices that convert mechanical energy to electrical power")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:48]: Generators are devices that convert mechanical energy to electrical power

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> Generators { get; set; } = new List<IBHoMObject>();
+
+        [Description("MEP Lighting is inclusive of all light fixtures")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:51]: LightFixtures includes devices that provide and distribute illumination

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+        public virtual List<IBHoMObject> LightFixtures { get; set; } = new List<IBHoMObject>();
+
+        [Description("List of BHoM WireSegments MEP Wiring is a flexible conductor of electricity")]
```
Comment (ajensen19) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:54]: Wiring includes the system of distribution cabling that conveys electricity

**Review by ajensen19 — APPROVED** [review]
Looks good! The main changes were making each category clear that it is meant to capture many incoming elements (this category "includes"...). And getting rid of MEP before stuff because I think that didn't make sense once the groups were broken out. It's one of Emir's biggest pet peeves when I use MEP where I should use mechanical, electrical or plumbing individually. That's my two cents!

My last thought was whether we need a separate electrical line item for solar panels. At the moment I'd think it would fall in equipment but could be good to break that one out if its not too much trouble. Can't wait for the day we try to include the embodied carbon of solar in a NZE building :)

File: LifeCycleAssessment_oM/Enums/ObjectCategory.cs [hunk 15]
```diff
@@ -44,10 +45,11 @@ public enum ObjectCategory
         GradeBeam,
         InteriorDoor,
         InteriorGlazing,
-        Lighting,
+        LightFixtures,
         PartitionWall,
         Pile,
-        Piping,    
+        Pipes,
+        Sprinklers,
```
Comment (FraserGreenroyd) [inline LifeCycleAssessment_oM/Enums/ObjectCategory.cs:52]: `Sprinklers` should come after `Slab`

File: LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs (hunk 8 above)
Comment (FraserGreenroyd) [inline LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:36]: `Batties` should be `Batteries`?

Comment (FraserGreenroyd) [reply LifeCycleAssessment_oM/ScopeObjects/ElectricalScope.cs:36]: Also `eg` should be `e.g.`

File: LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs (hunk 4 above)
Comment (FraserGreenroyd) [inline LifeCycleAssessment_oM/ScopeObjects/PlumbingScope.cs:35]: Just `a` machine?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Couple of things to fix, rest I can live with as @ajensen19 is happy.

**Comment by michaelhoehn** [comment]
> My last thought was whether we need a separate electrical line item for solar panels. At the moment I'd think it would fall in equipment but could be good to break that one out if its not too much trouble. Can't wait for the day we try to include the embodied carbon of solar in a NZE building :)

Happy to add this category in another PR as the MEP scope objects will be changing again this milestone. The goal of this PR is to provide a more logical framework for the workshops that will be taking place in subsequent sprints. Be sure to input during those workshops @ajensen19 😄

**Review by FraserGreenroyd — APPROVED** [review]
LGTM now

**Pull request merged**